### DISCUSSION TITLE
Vue 组件创建与使用

### DISCUSSION THREAD

**Discussion by zhmushan** [design-to-release/cosmic-design#14] [General]
其他问题出现再补充，仅组件库目前存在的问题，做以下建议：

### 不使用 `<script setup>`

`setup` 语法糖会将顶层定义的所有变量都暴露出去

### export CFoo 而不是 Foo

便于 `kebab-case` 使用，见下文“使用 `kebab-case` 而不是 Pascal”

### `kebab-case` 而不是 Pascal

在 HTML 中，`<Button>` 等价于 `<button>`，在模版中使用组件时，应遵循 HTML 直觉。（这一个理由就够了，别的不详细展开）

**Comment by zhmushan** [comment]
<img width="1061" alt="2b8aafad651fafc43075a511d41de473" src="https://user-images.githubusercontent.com/24505451/154907638-37c66fdc-8dec-42de-b264-6cb6075f5d88.png">

**Comment by qiansc** [comment]
` <script setup> setup 语法糖会将顶层定义的所有变量都暴露出去`
经过线下沟通和实际的测试：

* 变量暴露不存在
* import 编译后会提取出去
* const 部分会直接写入或者保留在setup，在组件创建过程可能常量部分有冗余开销，参考楼上图示方案拆块。

使用 setup 的原因是希望提升开发体验，减少开发时的冗余代码，性能暂时不是第一考虑，后续通过其他手段优化或者JSX方案。
